drivers/usb/usbport/device.c

```diff
@@ -1502,6 +1622,40 @@ USBPORT_RemoveDevice(IN PDEVICE_OBJECT FdoDevice,
         USBPORT_FreeUsbAddress(FdoDevice, DeviceHandle->DeviceAddress);
     }
 
+    if (!IsListEmpty(&DeviceHandle->TtList))
```

Comment:
Rewriting this as `while (!IsListEmpty(...))` would change the print logic. I guess the print is not worth this mess.

Reply:
It should not if you do:
```
if (!IsListEmpty(&DeviceHandle->TtList))
    // The DPRINT

while (!IsListEmpty(&DeviceHandle->TtList))
{
    // Contents of the original do-while loop.
}
```

Reply:
Right, that just seems silly ;p